github/login.go

```diff
@@ -83,3 +92,19 @@ func validateResponse(user *github.User, resp *github.Response, err error) error
 	}
 	return nil
 }
+
+func githubClientFromAuthURL(authURL string, httpClient *http.Client) (*github.Client, error) {
+	if strings.HasPrefix(authURL, "https://github.com/") {
```

Comment:
It worries me some we can't have the regular Github Client as the default (else) behavior. I see why you've got it this way since GithubEnterprise auth url's don't look likely to have any structure we could detect and leverage.

I think everyone using gologin with Github today should have an Endpoint.AuthURL starting with `https://github.com` (Github forbits http), but there may be some cases someone proxies access through another endpoint.

Reply:
How about adding a `func EnterpriseCallbackHandler(config *oauth2.Config, success, failure http.Handler)` method that is the same as `CallbackHandler`, but sets an enterprise API root URL?

Reply:
I'd rather not. Exposing the function would make Github Enterprise a first-class use case. Then if Github's enterprise product were to make a breaking change, I'd be expected to accomodate that in gologin.

As is, this PR would just let Github login automatically best-effort allow GHE (undocumented). If it becomes problematic, it can be reverted.

Reply:
Taking a step back, if a users wants a handler for GHE, it would be very much in the gologin style to just add convenience functions that chain [gologin/oauth2](https://github.com/dghubble/gologin/tree/master/oauth2) with a `EnterpriseCallbackHandler`. And that handler can go in their code or here.

It really comes down to whether GHE should be in-scope for gologin or left to users.

Reply:
Alright, I think your `EnterpriseCallbackHandler` is a reasonable approach. It eliminates the risk this will break anyone using regular Github login and it will be intuitive - its clear and isolated and avoids any magic of having GHE handling in the Github handler. And gologin is somewhat insulated from GHE making drastic breakages since the go-github library would have to deal with them first.

Ignore my "I'd rather not" from earlier.